draft-ietf-quic-qpack.md

```diff
@@ -369,60 +369,72 @@ name and same value).  Therefore, duplicate entries MUST NOT be treated as an
+less than or equal to the new table capacity.  This mechanism can be used to
+completely clear entries from the dynamic table by setting a capacity of 0,
+which can subsequently be restored.
+
 
-Whenever the maximum size for the dynamic table is reduced by the encoder,
-entries are evicted from the end of the dynamic table until the size of the
-dynamic table is less than or equal to the new maximum size.  This mechanism can
-be used to completely clear entries from the dynamic table by setting a maxiumum
-size of 0, which can subsequently be restored.
+### Maximum Dynamic Table Capacity
 
+To bound the memory requirements of the decoder, the value that the encoder is
+permitted to set as dynamic table capacity is limited.  In HTTP/3, this value is
+determined by the SETTINGS_MAXIMUM_DYNAMIC_TABLE_CAPACITY setting sent by the
+decoder (see {{configuration}}).  The encoder MUST not set a dynamic table
+capacity that exceeds this maximum, but it can choose to use a lower dynamic
+table capacity (see {{capacity-update}}).
 
-### Maximum Table Size
 
-The encoder decides how to update the dynamic table size and as such can control
-how much memory is used by the dynamic table.  To limit the memory requirements
-of the decoder, the dynamic table size is strictly bounded.  The decoder
-determines the maximum size that the encoder is permitted to set for the dynamic
-table.  In HTTP/3, this value is determined by the SETTINGS_HEADER_TABLE_SIZE
-setting (see {{configuration}}).  The encoder MUST not set a dynamic table size
-that exceeds this maximum, but it can choose to use a lower dynamic table size
-(see {{size-update}}).
+### Initial Dynamic Table Capacity
 
-The initial maximum size is determined by the corresponding setting when HTTP
-requests or responses are first permitted to be sent. For clients using 0-RTT
-data in HTTP/3, the table size is the remembered value of the setting, even if
-the server later specifies a larger maximum in its SETTINGS frame.  For HTTP/3
-servers and HTTP/3 clients when 0-RTT is not attempted or is rejected, the
-initial maximum table size is the value of the setting in the peer's SETTINGS
-frame.
+The initial dynamic table capacity is the value of the maximum dynamic table
+capacity when HTTP requests or responses are first permitted to be sent.  For
+HTTP/3 servers and HTTP/3 clients when 0-RTT is not attempted or is rejected,
+the initial maximum table capacity is zero.  For clients using 0-RTT data in
+HTTP/3, the initial table capacity is the remembered value of the setting, even
+if the server later specifies a different maximum dynamic table capacity in its
+SETTINGS frame.  It is not an error for the SETTINGS frame to contain a
+SETTINGS_MAXIMUM_DYNAMIC_TABLE_CAPACITY value lower than the initial table
+capacity on a 0-RTT connection; the maximum dynamic table capacity only limits
+subsequent table capacity changes.  However, if an endpoint receives a
```

Comment:
This is so that a decoder can request reduction of maximum dynamic table capacity on a new connection.

There are other, much simpler options:
1. Ban an endpoint from sending a SETTINGS_MAXIMUM_DYNAMIC_TABLE_CAPACITY value on a 0-RTT connection that is smaller than the remembered value.
2. Have the maximum table capacity always default to 0, even on a 0-RTT connection, until the SETTINGS frame is received.

Reply:
HTTP/3 in general bars reducing the value of a setting while accepting 0-RTT, so (1) would be more consistent.  From 4.2.5.2 in H3:

> If 0-RTT data is accepted by the server, its SETTINGS frame MUST NOT reduce any limits or alter any values that might be violated by the client with its 0-RTT data.

Reply:
Yes, just forbid reductions.  It's simpler.

I don't like this because I think that it means that I have to send a capacity update instruction, which I don't currently, but that's my problem.